web/src/app/components/bar-chart/bar-chart.component.ts

```diff
@@ -436,9 +463,109 @@ export class BarChartComponent implements AfterViewInit {
       .extent([[70, 0], [this.width - 90, this.heightBrush]]) // Limit the brush.
       .on('brush', brushDown); // Update the focus bar chart based on the brush selection.
 
-    this.brush.append('g')
+    this.brush
+      .append('g')
       .attr('class', 'brush')
       .call(brushSelector)
       .call(brushSelector.move, [firstItemPosition, lastItemPosition]);
   }
+
+  /**
+   * This function creates a boxplot next to the focus bar chart with all selected data.
+   * It indicates maximum, minimium, median, first quantile and third quantile of the
+   * selected data durations.
+   *
+   * @param inputData: Data selected for the focus bar chart
+   */
+  private createBoxplot(inputData: Item[]): void {
+    if (!this.boxplot) { return; }
+    // Remove all elements of from the previous boxplot.
+    this.boxplot.selectAll('circle').remove();
+    this.boxplot.selectAll('line').remove();
+    this.boxplot.selectAll('text').remove();
+
+    const durationSelected = inputData.map((obj: Item) => obj.durationHours);
+    const durationSorted = durationSelected.sort(d3.ascending);
+    const durationMax = durationSorted[durationSorted.length - 1];
+    const durationMin = durationSorted[0];
+
+    const q1 = d3.quantile(durationSorted, 0.25);
+    if (!q1) { return; }
+    const median = d3.quantile(durationSorted, 0.5);
+    if (!median) { return; }
+    const q3 = d3.quantile(durationSorted, 0.75);
+    if (!q3) { return; }
+    const dataLabels = [durationMax, durationMin, q1, median, q3];
+
+    const boxWidth = 30;
+    // If more than 100 Items are selected, set the circle radius to be 1.4; otherwise, set
```

Comment:
Prefer to say "If more than DEFAULT_MAX_LABELS of Items are selected...", because we don't know where the 100 comes from.